apis/database/v1alpha1/redshift_types.go

```diff
@@ -0,0 +1,713 @@
+
+	// FinalClusterSnapshotRetentionPeriod is the number of days that
+	// a manual snapshot is retained.
+	// If the value is -1, the manual snapshot is retained indefinitely.
+	// The value must be either -1 or an integer between 1 and 3,653.
+	// default -1
+	// +optional
+	FinalClusterSnapshotRetentionPeriod *int64 `json:"finalClusterSnapshotRetentionPeriod,omitempty"`
+
+	// HSMClientCertificateIdentifier specifies the name of the HSM client certificate
+	// the Amazon Redshift cluster uses to retrieve the data encryption keys stored in an HSM.
+	// +optional
+	HSMClientCertificateIdentifier *string `json:"hsmClientCertificateIdentifier,omitempty"`
+
+	// HSMConfigurationIdentifier specifies the name of the HSM configuration that
+	// contains the information the Amazon Redshift cluster can use to retrieve
+	// and store keys in an HSM.
+	// +optional
+	HSMConfigurationIdentifier *string `json:"hsmConfigurationIdentifier,omitempty"`
+
+	// IAMRoles is a list of AWS Identity and Access Management (IAM) roles that can be used
+	// by the cluster to access other AWS services. You must supply the IAM roles
+	// in their Amazon Resource Name (ARN) format. You can supply up to 10 IAM roles
+	// in a single request.
+	// A cluster can have up to 10 IAM roles associated with it at any time.
+	// kubebuilder:validation:MaxItems=10
+	// +optional
+	IAMRoles []string `json:"iamRoles,omitempty"`
+
+	// KMSKeyID for an encrypted DB cluster.
+	// The KMS key identifier is the Amazon Resource Name (ARN) for the KMS encryption
+	// key. If you are creating a DB cluster with the same AWS account that owns
+	// the KMS encryption key used to encrypt the new DB cluster, then you can
+	// use the KMS key alias instead of the ARN for the KM encryption key.
+	// +optional
+	KMSKeyID *string `json:"kmsKeyID,omitempty"`
+
+	// MaintenanceTrackName an optional parameter for the name of the maintenance track for the cluster.
+	// +optional
+	MaintenanceTrackName *string `json:"maintenanceTrackName,omitempty"`
+
+	// The default number of days to retain a manual snapshot. If the value is -1,
+	// the snapshot is retained indefinitely. This setting doesn't change the retention
+	// period of existing snapshots.
+	// default=1
+	// +kubebuilder:validation:Maximum=3653
+	// +optional
+	ManualSnapshotRetentionPeriod *int64 `json:"manualSnapshotRetentionPeriod,omitempty"`
+
+	// MasterUserPassword is the password associated with the master user account
+	// for the cluster that is being created.
+	// Constraints:
+	//    * Must be between 8 and 64 characters in length.
+	//    * Must contain at least one uppercase letter.
+	//    * Must contain at least one lowercase letter.
+	//    * Must contain one number.
+	//    * Can be any printable ASCII character (ASCII code 33 to 126) except '
+	//    (single quote), " (double quote), \, /, @, or space.
+	// +optional
+	NewMasterUserPassword *string `json:"masterUserPassword,omitempty"`
```

Comment:
Might be nice to add some notes about default behavior here if this field is not supplied.

Reply:
`NewMasterUserPassword` is for when the user wants to update or change the password generated by the controller at time of the cluster creation. So there is no default behavior to this. If you want to update the existing password you supply the value to it, else the password remains the same to what it was at creation.